x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -396,8 +401,10 @@ export function deleteFeatureFromIndex(featureId: string) {
       return;
     }
     try {
+      dispatch(updateEditShape(DRAW_SHAPE.WAIT));
       await (layer as IVectorLayer).deleteFeature(featureId);
       await dispatch(syncDataForLayerDueToDrawing(layer));
+      dispatch(updateEditShape(DRAW_SHAPE.DELETE));
```

Comment:
Can you move `dispatch(updateEditShape(DRAW_SHAPE.DELETE));` to after the try/catch. If deleteFeature fails, then edit mode stays in WAIT.